homeassistant/components/elevenlabs/config_flow.py

```diff
@@ -0,0 +1,205 @@
+            )
+        # Validate auth, get voices
+        try:
+            self.voices, self.models = await get_voices_models(user_input[CONF_API_KEY])
+        except ApiError:
+            errors[CONF_API_KEY] = "ElevenLabs API responded with an Error!"
+        if errors:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA_NO_AUTH, errors=errors
+            )
+
+        self.user_info = user_input
+        return await self.async_step_voice()
+
+    async def async_step_voice(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the voice selection step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="voice", data_schema=self._get_user_schema_authenticated()
+            )
+        # Add api_key to user input
+        assert self.user_info is not None
+        user_input[CONF_API_KEY] = self.user_info[CONF_API_KEY]
+
+        return self.async_create_entry(
+            title=self.models[user_input[CONF_MODEL]], data=user_input
+        )
+
+    @staticmethod
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlow:
+        """Create the options flow."""
+        return ElevenLabsOptionsFlow(config_entry)
+
+
+class ElevenLabsOptionsFlow(OptionsFlow):
+    """ElevenLabs options flow."""
+
+    # id -> name
+    voices: dict[str, str] = {}
+    models: dict[str, str] = {}
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.api_key: str = self.config_entry.data[CONF_API_KEY]
+        self.voices_models_loaded = False
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options."""
+        if not self.voices_models_loaded:
+            self.voices, self.models = await get_voices_models(self.api_key)
+            self.voices_models_loaded = True
+
+        assert self.models and self.voices
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.models[user_input[CONF_MODEL]], data=user_input
+            )
+
+        options: dict[str, str] = {
+            CONF_MODEL: self.config_entry.options.get(
+                CONF_MODEL, self.config_entry.data[CONF_MODEL]
+            ),
+            CONF_VOICE: self.config_entry.options.get(
+                CONF_VOICE, self.config_entry.data[CONF_VOICE]
+            ),
+        }
+
+        schema = self.elevenlabs_config_option_schema(options)
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(schema),
+        )
+
+    def elevenlabs_config_option_schema(self, options: dict[str, str]) -> dict:
+        """Elevenlabs options schema."""
+        return {
+            vol.Required(
+                CONF_MODEL,
```

Comment:
I don't see translations for these options in `strings.json`

Reply:
Added them.